services/QuillLMS/client/app/bundles/Connect/components/activityHealth/activityHealth.tsx

```diff
@@ -0,0 +1,417 @@
+        Filter: ({ filter, onChange }) =>
+        (
+          <NumberFilterInput
+            filter={filter}
+            handleChange={onChange}
+            label="Filter for average difficulty"
+          />
+        ),
+        resizeable: true,
+        sortMethod: sort,
+        Cell: props => props.value,
+        maxWidth: 150
+      },
+      {
+        Header: 'Standard Deviation Difficulty',
+        accessor: 'standard_dev_difficulty',
+        filterMethod: filterNumbers,
+        Filter: ({ filter, onChange }) =>
+        (
+          <NumberFilterInput
+            filter={filter}
+            handleChange={onChange}
+            label="Filter for recent plays"
+          />
+        ),
+        resizeable: true,
+        sortMethod: sort,
+        Cell: props => props.value,
+        maxWidth: 150
+      },
+      {
+        Header: 'Average Common Unmatched',
+        accessor: 'avg_common_unmatched',
+        filterMethod: filterNumbers,
+        Filter: ({ filter, onChange }) =>
+        (
+          <NumberFilterInput
+            filter={filter}
+            handleChange={onChange}
+            label="Filter for avg common unmatched"
+          />
+        ),
+        resizeable: true,
+        sortMethod: sort,
+        Cell: props => props.value,
+        maxWidth: 150
+      }
+    ];
+  }
+
+  addCommasToThousands = (num) =>
+  {
+    if (!num) return ""
+    var num_parts = num.toString().split(".");
+    num_parts[0] = num_parts[0].replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+    return num_parts.join(".");
+  }
+
+  getToolBadge = (tool) => {
+    if (tool === CONNECT_TOOL) {
+      return (<div className="badge connect-badge">c</div>)
+    } else {
+      return (<div className="badge grammar-badge">g</div>)
+    }
+  }
+
+  formatTableForCSV = (e) => {
+    const currentRecords = this.reactTable.getResolvedState().sortedData
+    const columns = this.columnDefinitions()
+    let dataToDownload = []
+    for (var index = 0; index < currentRecords.length; index++) {
+       let recordToDownload = {}
+       for(var colIndex = 0; colIndex < columns.length ; colIndex ++) {
+          recordToDownload[columns[colIndex].Header] = currentRecords[index][columns[colIndex].accessor]
+       }
+       dataToDownload.push(recordToDownload)
+    }
+
+    // Map Activity Packs to strings because JSON objects dont display in CSVs
+    let clonedDataToDownload = JSON.parse(JSON.stringify(dataToDownload));
+    clonedDataToDownload.forEach(item=> {
+      item["Activity Packs"] = item["Activity Packs"] ? item["Activity Packs"].map(v => v.name) : ''
+    });
+
+    this.setState({ dataToDownload: clonedDataToDownload }, () => {
+       this.csvLink.link.click();
+    })
+  }
+
+  handleFlagChange = (e) => {
+    const { dispatch } = this.props
+    dispatch(actions.setFlag(e.target.value))
+
+    // this is a dummy action, we dont use the state value but we need to trigger a re-render
+    this.setState({flag: e.target.value})
```

Comment:
If you didn't have that `shouldComponentUpdate` method above, you wouldn't need this dummy method here because the prop change should trigger a re-render.